File: 2025-09-25 - Lista Prioridade PQ/Exemplo_Python/TesteListaPrioridadePQ.py
# ------------------------------------------------------------------------
# testeListaPrioridadePQ.py
# Programa principal para demonstrar todas as operações da lista de prioridade
# ------------------------------------------------------------------------
from Node import Node
from ListaPrioridadePQ import ListaPrioridadePQ

def main():
    lista = ListaPrioridadePQ()

    # -----------------------------------------------------------------
    # Inserção de elementos
    # -----------------------------------------------------------------
    lista.inserir(Node("TEC.0007", "Programação de Computadores", 120, 1))
    lista.inserir(Node("TEC.0020", "Estrutura de Dados Lineares", 80, 2))
    lista.inserir(Node("TEC.0027", "Estrutura de Dados Não-Lineares", 80, 3))
    lista.inserir(Node("TEC.0142", "Programação para Redes", 120, 3))
    lista.inserir(Node("TEC.0022", "Bancos de Dados", 80, 1))

    # -----------------------------------------------------------------
    # Imprimir lista
    # -----------------------------------------------------------------
    lista.imprimirLista()
    print("Tamanho da lista:", lista.tamanho())
    print("--------------------------------------")

    # -----------------------------------------------------------------
    # Buscar elemento
    # -----------------------------------------------------------------
    busca = lista.buscar("TEC.0020")
    if busca:
        print("Disciplina encontrada:", busca.nomeDisciplinaProp)
    else:
        print("Disciplina não encontrada.")
    print("--------------------------------------")

    # -----------------------------------------------------------------
    # Atualizar disciplina
    # -----------------------------------------------------------------
    lista.atualizar(Node("TEC.0020", "Estrutura de Dados Lineares Avançada", 100, 2))
    lista.imprimirLista()
    print("Tamanho da lista após atualização:", lista.tamanho())
    print("--------------------------------------")

    # -----------------------------------------------------------------
    # Remover disciplina
    # -----------------------------------------------------------------
    lista.remover("TEC.0007")
    lista.imprimirLista()
    print("Tamanho da lista após remoção:", lista.tamanho())

if __name__ == "__main__":
    main()

File: 2025-09-25 - Lista Prioridade PQ/Exemplo_C#/TesteListaPrioridadePQ.cs
using System;

// ------------------------------------------------------------------------
// TesteListaPrioridadePQ.cs
// Programa principal para demonstrar todas as operações da lista de prioridade
// ------------------------------------------------------------------------
class TesteListaPrioridadePQ
{
    static void Main(string[] args)
    {
        ListaPrioridadePQ lista = new ListaPrioridadePQ();

        // -----------------------------------------------------------------
        // Inserção de elementos
        // -----------------------------------------------------------------
        lista.inserir(new Node("TEC.0007", "Programação de Computadores", 120, 1));
        lista.inserir(new Node("TEC.0020", "Estrutura de Dados Lineares", 80, 2));
        lista.inserir(new Node("TEC.0027", "Estrutura de Dados Não-Lineares", 80, 3));
        lista.inserir(new Node("TEC.0142", "Programação para Redes", 120, 3));
        lista.inserir(new Node("TEC.0022", "Bancos de Dados", 80, 1));

        // -----------------------------------------------------------------
        // Imprimir lista
        // -----------------------------------------------------------------
        lista.imprimirLista();
        Console.WriteLine("Tamanho da lista: " + lista.tamanho());
        Console.WriteLine("--------------------------------------");

        // -----------------------------------------------------------------
        // Buscar elemento
        // -----------------------------------------------------------------
        Node busca = lista.buscar("TEC.0020");
        if (busca != null)
        {
            Console.WriteLine("Disciplina encontrada: " + busca.nomeDisciplinaProp);
        }
        else
        {
            Console.WriteLine("Disciplina não encontrada.");
        }
        Console.WriteLine("--------------------------------------");

        // -----------------------------------------------------------------
        // Atualizar disciplina
        // -----------------------------------------------------------------
        lista.atualizar(new Node("TEC.0020", "Estrutura de Dados Lineares Avançada", 100, 2));
        lista.imprimirLista();
        Console.WriteLine("Tamanho da lista após atualização: " + lista.tamanho());
        Console.WriteLine("--------------------------------------");

        // -----------------------------------------------------------------
        // Remover disciplina
        // -----------------------------------------------------------------
        lista.remover("TEC.0007");
        lista.imprimirLista();
        Console.WriteLine("Tamanho da lista após remoção: " + lista.tamanho());
    }
}
